mapadroid/worker/WorkerQuests.py

```diff
@@ -710,15 +727,24 @@ def _handle_stop(self, timestamp: float):
                     return
             elif (type_received == LatestReceivedType.FORT_SEARCH_RESULT
                     and data_received == FortSearchResultTypes.FULL):
-                self.logger.warning("Failed getting quest but got items - quest box is probably full. Starting cleanup "
-                                    "routine.")
-                reached_main_menu = self._check_pogo_main_screen(10, True)
-                if not reached_main_menu:
+                self._got_items_failcount += 1
+                if self._got_items_failcount < 3:
+                    self.logger.warning(f"Failed getting quest but got items ({self._got_items_failcount}) - quest "
+                                        "box is probably full. Starting cleanup routine.")
+                    reached_main_menu = self._check_pogo_main_screen(10, True)
+                    if not reached_main_menu:
+                        if not self._restart_pogo(mitm_mapper=self._mitm_mapper):
+                            # TODO: put in loop, count up for a reboot ;)
```

Comment:
method restart_pogo actually reboots device in case restarting fails, so this todo should be removed